src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -21,4 +21,24 @@ object PeerGroup {
   trait TerminalPeerGroup[A, F[_]] extends PeerGroup[A, F]
 
   case class InitializationError(message: String, cause: Throwable)
+
+  def create[PG](pg: => PG, config: Any): Either[InitializationError, PG] =
+    try {
+      Right(pg)
+    } catch {
+      case t: Throwable =>
+        Left(InitializationError(initializationErrorMsg(config), t))
+    }
+
+  def createOrThrow[PG](pg: => PG, config: Any): PG =
+    try {
+      pg
+    } catch {
+      case t: Throwable =>
+        throw new IllegalStateException(initializationErrorMsg(config), t)
+    }
+
+  private def initializationErrorMsg[F[_]](config: Any) =
+    s"Failed initialization of ${classOf[TCPPeerGroup[F]].getName} with config $config. Cause follows."
```

Comment:
Have you tried the usefulness of printing `Config`? Sometimes it can be just too noisy without the actual details needed to debug, it could print a useless string too.